README.rst

```diff
@@ -10,4 +10,174 @@ test base classes and documentation shared by many XBlocks.
+* Float:
+  ``field_name = Float(display_name="Field Name")`` 
+* Integer:
+  ``field_name = Integer(display_name="Field Name")`` 
+* String:
+  ``field_name = String(display_name="Field Name")`` 
+* String (multiline):
+  ``field_name = String(multiline_editor=True, resettable_editor=False)``
+* String (html):
+  ``field_name = String(multiline_editor='html', resettable_editor=False)``
+* Any of the above will use a dropdown menu if they have a pre-defined
+  list of possible values.
+* List of undordered unique values (i.e. sets) drawn from a small set of
+  possible values:
+  ``field_name = List(list_style='set', list_values_provider=some_method)``
+
+  - The ``List`` declaration must include the property ``list_style='set'`` to
+    indicate that the ``List`` field is being used with set semantics.
+  - The ``List`` declaration must also define a ``list_values_provider`` method
+    which will be called with the block as its only parameter and which must
+    return a list of possible values.
+* Rudimentary support for List, Dict, and any other JSONField-derived field types
+
+  - ``list_field = List(display_name="Normal List", default=[])``
+  - ``dict_field = Dict(display_name="Normal Dict", default={})``
+
+Supported field options (all field types):
+
+* ``values`` can define a list of possible options, changing the UI element
+  to a select box. Values can be set to any of the formats `defined in the
+  XBlock source code <https://github.com/edx/XBlock/blob/master/xblock/fields.py>`__:
+  
+  - A finite set of elements: ``[1, 2, 3]``
+  - A finite set of elements where the display names differ from the values
+    ::
+        [
+            {"display_name": "Always", "value": "always"},
+            {"display_name": "Past Due", "value": "past_due"},
+        ]
+  - A range for floating point numbers with specific increments:
+    ``{"min": 0 , "max": 10, "step": .1}``
+  - A callable that returns one of the above. (Note: the callable does
+    *not* get passed the XBlock instance or runtime, so it cannot be a
+    normal member function)
+* ``values_provider`` can define a callable that accepts the XBlock
+  instance as an argument, and returns a list of possible values in one
+  of the formats listed above.
+* ``resettable_editor`` - defaults to ``True``. Set ``False`` to hide the
+  "Reset" button used to return a field to its default value by removing
+  the field's value from the XBlock instance.
+
+Basic screenshot: |Screenshot 1|
+
+StudioContainerXBlockMixin
+--------------------------
+
+.. code:: python
+
+    from xblockutils.studio_editable import StudioContainerXBlockMixin
+
+This mixin helps with creating XBlocks that want to allow content
+authors to add/remove/reorder child blocks. By removing any existing
+``author_view`` and adding this mixin, you'll get editable,
+re-orderable, deletable child support in Studio. To enable authors to
+add new children, simply override ``author_edit_view`` and set
```

Comment:
Ok, I can see this needs re-wording. These sentences were more about whether or not we want to allow authors to add arbitrary *types* of child XBlocks (and not about allowing or denying an arbitrary *number* of XBlocks). I've reworded it in my upcoming commit.